thunderfish/lizpulses.py

```diff
@@ -120,22 +144,32 @@ def extract_eod_times(data, peakwidth, cutwidth, threshold_factor=2):
     """
     
     threshold = np.mean(np.abs(data))*threshold_factor
-    x_peaks, x_troughs = detect_peaks(data, threshold)
+    orig_x_peaks, orig_x_troughs = detect_peaks(data, threshold)
 
-    if len(x_peaks)==0:
+    if len(orig_x_peaks)==0:
+        if verbose>0:
+            print('No peaks detected.')
         return [], [], [], []
     else:
-        peaks = makeeventlist(x_peaks, x_troughs, data, peakwidth)
+        peaks = makeeventlist(orig_x_peaks, orig_x_troughs, data, peakwidth)
         peakindices, _, _ = discardnearbyevents(peaks[0], peaks[3], peakwidth)
         x_peaks, x_troughs, eod_hights, eod_widths = discard_connecting_eods(peaks[0][peakindices], peaks[1][peakindices], peaks[3][peakindices], peaks[4][peakindices])
         
         # only take those where the cutwidth does not casue issues
         cut_idx = np.where((x_peaks>int(cutwidth/2)) & (x_peaks<(len(data)-int(cutwidth/2))) & (x_troughs>int(cutwidth/2)) & (x_troughs<(len(data)-int(cutwidth/2))))[0]
+        
+        if verbose>0:
+            print('Peaks detected:                        %i'%len(orig_x_peaks))
+            print('Sidepeaks discarded:                   %i'%(len(orig_x_peaks)-len(peakindices)))
+            print('Connecting eods discarded:             %i'%(len(peakindices)-len(x_peaks)))
+            print('Peaks discarded due to cutwidth size:  %i'%(len(x_peaks)-len(cut_idx)))
+            print('                                       ----')
+            print('Total amount of EOD peaks:             %i'%(len(cut_idx)))

```

Comment:
Nice, this is the information that is helpful to see what is going on with the data. Maybe some of this can go into the respective functions. But some kind of summary is nice (number of EOD peaks and number of clusters) here.